File: frontend/src/lib/transport.ts
/**
 * Transport encoding for sensitive payloads.
 * Encodes data before sending over the network so it's not human-readable in DevTools.
 * The backend decodes on receipt and encodes responses.
 */

export function encodePayload(data: string): string {
  if (!data) return "";
  // Double base64 + reverse to make it unreadable
  const first = btoa(unescape(encodeURIComponent(data)));
  const reversed = first.split("").reverse().join("");
  return btoa(reversed);
}

export function decodePayload(encoded: string): string {
  if (!encoded) return "";
  try {
    const reversed = atob(encoded);
    const first = reversed.split("").reverse().join("");
    return decodeURIComponent(escape(atob(first)));
  } catch {
    return encoded; // Return as-is if not encoded (legacy data)
  }
}

File: backend/transport.py
"""Transport encoding for sensitive payloads.

Encodes/decodes data so it's not human-readable in browser DevTools Network tab.
Uses double base64 + reverse to obfuscate.
"""

import base64


def encode_payload(data: str) -> str:
    """Encode a string for transport (matches frontend encodePayload)."""
    if not data:
        return ""
    first = base64.b64encode(data.encode("utf-8")).decode("ascii")
    reversed_str = first[::-1]
    return base64.b64encode(reversed_str.encode("ascii")).decode("ascii")


def decode_payload(encoded: str) -> str:
    """Decode a transport-encoded string (matches frontend decodePayload)."""
    if not encoded:
        return ""
    try:
        reversed_str = base64.b64decode(encoded).decode("ascii")
        first = reversed_str[::-1]
        return base64.b64decode(first).decode("utf-8")
    except Exception:
        return encoded  # Return as-is if not encoded
